inst/rmd/qc_report.rmd

```diff
@@ -420,7 +423,87 @@ if(has_processed){
+The above plot shows the UMAP (Uniform Manifold Approximation and Projection) embeddings for each cell, coloring each cell by the total number of genes detected per cell.
+The below plots show the same UMAP embeddings, coloring each cell by the gene expression of the gene indicated in the label for each panel. The genes chosen for plotting are the top 12 highly variable genes identified in the library.
+
+```{r message=FALSE}
+if(has_processed){
+  
+  if(has_var_genes){
+    
+    # select top genes to plot 
+    top_genes <- processed_meta$highly_variable_genes |>
+      head(n=12)
+    
+    # grab expression for top genes from counts
+    var_gene_exp <- logcounts(processed_sce[top_genes,]) |>
+      as.matrix() |>
+      t() |>
+      as.data.frame() |>
+      tibble::rownames_to_column("barcode")
+    
+    # grab rowdata as data frame to later combine with gene expression data
+    # rowdata contains the mapped gene symbol so can use that to label plots instead of ensembl gene id from rownames 
+    rowdata_df <- rowData(processed_sce) |>
+      as.data.frame() |>
+      tibble::rownames_to_column("ensembl_id") |>
+      select(ensembl_id, gene_symbol) |>
+      filter(ensembl_id %in% top_genes) |>
+      mutate(ensembl_id = factor(ensembl_id, levels = top_genes)) |>
+      arrange(ensembl_id) |>
+      mutate(gene_symbol = factor(gene_symbol, levels = gene_symbol))
+      
+    
+    # extract umap embeddings as a dataframe to join with gene expression and coldata for plotting
+    umap_df <- reducedDim(processed_sce, "UMAP") |>
+      as.data.frame() |>
+      tibble::rownames_to_column("barcode") |>
+      rename("UMAP1" = "V1",
+             "UMAP2" = "V2")
+    
+    # combine gene expression with coldata, umap embeddings, and rowdata and create data frame to use for plotting
+    coldata_df <- colData(processed_sce) |>
+      as.data.frame() |>
+      tibble::rownames_to_column("barcode") |>
+      # combine with gene expression
+      left_join(var_gene_exp, by = "barcode") |>
+      # combine with umap embeddings 
+      left_join(umap_df, by = "barcode") |>
+      # combine all genes into a single column for easy faceting 
+      tidyr::pivot_longer(cols = starts_with("ENSG"),
+                          names_to = "ensembl_id",
+                          values_to = "gene_expression") |>
+      # join with row data to add in gene symbols 
+      left_join(rowdata_df)
+    
+    
+  }
+  
+  ggplot(coldata_df, aes(x = UMAP1, y = UMAP2, color = gene_expression)) + 
+    geom_point(alpha = 0.1, size = 0.001) + 
+    facet_wrap(~ gene_symbol) +
+    scale_color_viridis_c() +
+    labs(
+      color = "Log-normalized \ngene expression"
+    ) +
+    theme(legend.position = "top", 
+          axis.text = element_text(size = 8, color = "black"),
+          axis.title = element_text(size = 8, color = "black"),
+          legend.title = element_text(size = 8),
+          legend.text = element_text(size = 6)) + 
```

Comment:
I probably wouldn't make these quite so small... 8 pt is already pretty small!

I wonder if we want to play with `strip.text` and `strip.background`. I think the grey background is okay, but we might go a bit smaller, or match the axis text, or no border on the background?

I might try:
```
strip.background = element_rect(fill = 'transparent'),
strip.text = element_text(size = 8) 
```

But not something to spend much time on.